src/jsonRpc.ts

```diff
@@ -287,8 +314,27 @@ function makeCodec(
         return
       }
       const pendingCall = remoteCalls.get(id)
+
+      // Handle subscription calls:
       if (pendingCall == null) {
-        sendError(-32603, `Cannot find id ${String(id)}`, id).catch(handleError)
+        const remoteSubscriptionMethodName = remoteSubscriptions.get(id)
+
+        if (remoteSubscriptionMethodName == null) {
+          sendError(-32603, `Cannot find id ${String(id)}`, id).catch(
+            handleError
+          )
+          return
+        }
+
+        const { asParams } = remoteCleaners[remoteSubscriptionMethodName]
+        const wasParams = uncleaner(asParams)
+        const payload = wasCall({
+          jsonrpc: '2.0',
+          method: remoteSubscriptionMethodName,
+          params: wasParams(result),
+          id: null // Notifications don't have ids
+        })
+        handleSend(JSON.stringify(payload)).catch(handleError)
```

Comment:
We should not be sending anything in here. Instead, we should be calling `localMethods[remoteSubscriptionMethodName](asParams( result ))

Reply:
I decided to call `handleMessage` recursively because it calls `localMethods` and does all the error checking for the message.